systests/cluster/test_remote_account.py

```diff
@@ -61,13 +64,326 @@ def write_to_log(self, msg):
         self.nodes[0].account.ssh("echo -e -n " + repr(msg) + " >> " + self.log_file)
 
 
-class RemoteAccountTest(Test):
+class GenericService(Service):
+    """Service which doesn't do anything - just a group of nodes each of which has a scratch directory."""
+    def __init__(self, context, num_nodes):
+        super(GenericService, self).__init__(context, num_nodes)
+        self.scratch_dir = "scratch"
+        for node in self.nodes:
+            node.account.mkdirs(self.scratch_dir)
+
+    def clean_node(self, node):
+        node.account.remove(self.scratch_dir, allow_fail=True)
+
+
+class FileSystemTest(Test):
+    """
+    Note that in an attempt to isolate the file system methods, validation should be done with ssh/shell commands.
+    """
+
+    def setup(self):
+        self.service = GenericService(self.test_context, 1)
+        self.node = self.service.nodes[0]
+        self.scratch_dir = self.service.scratch_dir
```

Comment:
I am confused about this test -- I thought `Test` was defining `scratch_dir` as a property so it would get created on demand but here we're overwriting it to be the fixed value in the `GenericService` above?